src/Nethermind/Nethermind.Serialization.Rlp/RlpFactory.cs

```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+using Nethermind.Core.Buffers;
+
+namespace Nethermind.Serialization.Rlp
+{
+    public class RlpFactory
```

Comment:
would it make sense to make it struct?

Reply:
Is class to have atomic writes on the `CappedArray<byte>` struct; which is `{byte[],int}` which otherwise could cause a concurrent reader to read the array but a different length from an earlier write for example; because the two items are not written atomicly.